sympy/discrete/tests/test_z_transforms.py

```diff
@@ -0,0 +1,249 @@
+"""
+test file for z_transforms and inverse_z_transforms
+"""
+from sympy.discrete.z_transforms import (z_transform,
+                                         inverse_z_transform,
+                                         z_pairs_properties,
+                                         z_pairs_prop_inverse,
+                                         round_float)
+from sympy.abc import z, n
+from sympy.core.symbol import symbols
+from sympy.core import S
+from sympy.functions.special.delta_functions import DiracDelta, Heaviside
+from sympy.functions.elementary.trigonometric import (cos, sin)
+from sympy.functions.elementary.complexes import Abs
+
+def test_z_transform():
+    """
+    test from z_transform table and excercises
+    http://blog.espol.edu.ec/telg1001/transformada-z-tabla/
+    """
+    a, b, c, = symbols('a, b, c', positive=True)
+    ZT = z_transform
+
+    # included in _z_pairs_table
+    assert ZT(DiracDelta(n),n,z) == 1
+    assert ZT(Heaviside(n),n,z) == z/(z - 1)
+    assert ZT(cos(a*n),n,z) == \
+           z*(z - cos(a))/(z**2 - 2*z*cos(a) + 1)
+    assert ZT(sin(a*n),n,z) == \
+           z*sin(a)/(z**2 - 2*z*cos(a) + 1)
+    assert ZT(cos(b*n+c),n,z) == \
+           z*(z*cos(c) - cos(b - c))/(z**2 - 2*z*cos(b) + 1)
+    # shift property
+    assert ZT(DiracDelta(n-a),n,z) == z**(-a)
+    assert ZT(DiracDelta(n+a),n,z) == z**a
+    assert ZT(b*DiracDelta(n+2),n,z) == b*z**2
+    assert ZT(-b*DiracDelta(n+2),n,z) == -b*z**2
+    assert ZT(Heaviside(n-a),n,z) == z**-a*z/(z - 1)
+    assert ZT(Heaviside(n+a),n,z) == z**a*z/(z - 1)
+    assert ZT(b*Heaviside(n+a),n,z) == b*z**a*z/(z - 1)
+    # multiply by n property
+    assert ZT(n*Heaviside(n),n,z) == z/(z - 1)**2
+    assert ZT(n**2*Heaviside(n),n,z) == z*(z + 1)/(z - 1)**3
+    assert ZT(n**3*Heaviside(n),n,z) == \
+           z*(z**2 + 4*z + 1)/(z - 1)**4
+    # multiply by a**n property
+    assert ZT(a**n*Heaviside(n),n,z) == z/(z - a)
+    assert ZT(0.5**n*Heaviside(n),n,z) == z/(z - 0.5)
+    assert ZT((S.One/2)**n*Heaviside(n),n,z) == 2*z/(2*z - 1)
+
+    # time reversal property
+    assert ZT(Heaviside(-n),n,z) == -1/(z - 1)
+    assert ZT(-(0.5**n)*Heaviside(-n-1),n,z) == z/(z - 0.5)
+
+    # combined properties
+    assert ZT(-n*(0.5**n)*Heaviside(-n-1),n,z) == 0.5*z/(z - 0.5)**2
```

Comment:
You should never use real numbers when you want rationals.  Write `S.Half` instead of `0.5`, or `S(3)/5` instead of `0.6` if it is supposed to be a rational number.

Reply:
greetings @hanspi42
Thanks for your comments to improve the algorithm, I would like to implement something about this one.
I test ZT with S.Half and Rational(3/4) and got this:
>>> ZT((S.Half)**n*Heaviside(n),n,z)
2*z/(2*z - 1)
>>> from sympy.core.numbers import Rational
>>> ZT((Rational(3/4))**n*Heaviside(n),n,z)
4*z/(4*z - 3)
the result should be expressed as rational number.
Should be included asserts with rational numbers as a first step.
I Appreciate your comments.

Reply:
This works by chance. If you write `3/4` in Python, it is immediately converted to `0.75`, handed over to `Rational`, and `Rational` converts it back to `3/4`. The correct way would be `Rational(3, 4)`. Try `Rational(15/123)` and `Rational(15, 123)`.

You can also do `S(3)/4`, because then Python will see a SymPy singleton object `S(3)` and will not convert anything to `0.75` when it is to be divided by `4`.